Common parent for OSS plugins

jdillon opened the pull request:
https://issues.sonatype.org/browse/NXCM-4924

Still pending how this might work in our pro plugins, or in thirdparty plugins... inheritance of a single parent to provide shared configuration is a real limitation... which means that some configuration will always have to be duplicated.

jdillon committed:
Seeing how a plugin-parent module might be used to simplify/unify plugin build configurations.

jdillon committed:
notes

jdillon committed:
more comments

jdillon committed:
more comments

jdillon committed:
kill comment

jdillon committed:
notes

jdillon committed:
Update pom, use maven 3 deps.
Do not include scope, for -compile.pom scope=compile is the default.  For -dependencies.pom, adding scope=provided to DM entries adds downstream complication to fix the scope as needed.
Some clean up.

jdillon committed:
Update provided-dependencies-maven-plugin to 1.6-SNAPSHOT.
Generate a -dependencies.pom for nexus-plugin-api so we can import it in plugin-parent.
More notes.

jdillon committed:
nexus-web-utils is provided by nexus-plugin-api

jdillon committed:
Forgot type=pom on scope=import

jdillon committed:
Use plugin-parent

jdillon committed:
kill cruft

jdillon committed:
Generate -dependencies.pom for nexus-plugin-test-api, include that in plugin-parent.
Add plexus-jetty-testsuite to nexus-plugin-test-api.
Update rrb plugin to use plugin-parent.

jdillon committed:
Add nexus.version property to plugin-parent.
Update mac plugin to use plugin-parent.

jdillon committed:
Update maven-bridge plugin to use plugin-parent.

jdillon committed:
Fix up groupIds

jdillon committed:
Use plugin-parent for restlet1x

jdillon committed:
Flip indexer bits to use plugin-parent

jdillon committed:
Trying to get ldap plugin using plugin-parent

jdillon commented:
Some hacks to sort out some issues with deps... might need a few more.  Trying to treat plugins/\* as isolated (in case they are pulled out into separate projects) but some of the DM configuration presently causes some duplication.

Also the provided-dependencies-maven-plugin only pays attention to deps defined in a project, not anything in DM which the project has access too, so in a few places had to add hard deps for things which were already in DM.

jdillon commented:
Once this is basically working in OSS, then we can update an external OSS plugin (likely capabilities) to use this parent for its project.  Then look at how to sort out the PRO side.  ATM I think for PRO we will be forced to duplicate most of plugin-parent/pom.xml content so that the PRO version of this can extend from the corporate pom and pick up all that configuration... so we should try to keep the plugin-parent/pom.xml as small as possible to make it manageable, and there is still more work to be done here once the basic concept is vetted and proven will work.

jdillon committed:
Another refinement of plugin-parent, avoid scope=import... its a fucking shit half-ass functional feature.

jdillon committed:
Bridge config for now

jdillon committed:
Use pluginName and pluginDescription which is the default expression the plugin looks for.

jdillon committed:
Use pluginName for now

jdillon committed:
Update enucniate version, set plugin url

jdillon committed:
More notes about the enunciate-core dep... I really hate this thing

jdillon committed:
Drop enunciate muck from top-level-pom, its all in plugin-parent now.
Various notes.
Flip off app-lifecycle in a few places where it probably does not belong... not sure what its there for.

jdillon committed:
update docs

jdillon committed:
drop import, this won't work for provided-dep*-m-p generated stuff in the same reactor execution anyways.

jdillon committed:
add DM for h2 plugin

jdillon committed:
Add DM for maven-bridge plugin

jdillon committed:
Add some sections

jdillon committed:
tidy

jdillon committed:
app-lifecycle does not appear to be needed here

jdillon committed:
flip app-lifecycle back on for its I think its needed here, clean up later.
Drop extra config in PM not needed.

jdillon committed:
Drop p-d-m-p use, will try and avoid scope=import due to complexity/problems it brings to the table.

jdillon committed:
plugin-api is picked up from nexus-core... hopefully will go away, so avoid referencing it directly.

jdillon committed:
make section markers a bit more obvious

jdillon committed:
Update  to use nexus-plugin-parent

jdillon committed:
add better message to help sort out WTF is going on

jdillon committed:
Use nexus-plugin-parent as parent of nexus-plugins (re-id from nexus-core-plugins)

jdillon committed:
Notes

jdillon committed:
Set pluginSiteUrl as property so it can be changed via property too

jdillon commented:
3 examples of use of nexus-plugin-parent: capabilities, yum, site-repository

jdillon committed:
Add maven.yui* default properties and optimize-js profile.

jdillon committed:
Add optimize-js profile to release configuration.

jdillon committed:
note why there is no <dependencies> or <plugins> configuration in this POM.

jdillon committed:
tidy docs

jdillon committed:
Bring over configuration of PRO plugin-parent

jdillon committed:
tidy

jdillon committed:
normalize.  Drop nexus-plugin-manger DM

jdillon committed:
update comments

jdillon committed:
re-order

jdillon committed:
tidy

jdillon committed:
Drop dependencies, this is only used by import

peterlynch commented on plugin-parent/pom.xml:
```diff
@@ -0,0 +1,522 @@
+  <parent>
+    <groupId>org.sonatype.forge</groupId>
+    <artifactId>forge-parent</artifactId>
+    <version>30</version>
```
31 is avail if you want

jdillon replied:
Okay, I'll update it.

jdillon committed:
use latest forge-parent

peterlynch commented:
looked over pretty close and didn't see anything to complain about

jdillon committed:
Use <nexus.version>${project.version}</nexus.version> in a few places where plugins are pulled in to legacy muck.
Doing this instead of DM since we don't want to promote that use and DM says its okay... and its not.

jdillon committed:
Drop commented sections

jdillon committed:
Merge branch 'master' into plugin-parent

Conflicts:
	plugins/ldap/nexus-ldap-plugin-it/pom.xml

jdillon committed:
Update headers

jdillon committed:
license plugin got confused

jdillon committed:
Drop DM for ancient restlight shit, they shouldn't be referenced by anything in this project.
Update deps for it helper plugin.

jdillon committed:
Drop, zion uses .zion covered by .* pattern

jdillon committed:
DM isn't very helpful once we've added plugin-parent

jdillon committed:
Do not build nexus-webapp-{compile|dependencies} this shouldn't be needed anymore with plugin-parent bits

jdillon committed:
Add PM for n-t-e-m-p

jdillon committed:
docs

jdillon committed:
add app-lifecycle, so nexus-plugin packaging is known

jdillon commented:
https://bamboo.zion.sonatype.com/browse/NXO-OSSF10-14 :white_check_mark:

jdillon committed:
Trim down deps for enunciate ResourceMethodSignature, we really only need this one class.

jdillon committed:
Tidy

jdillon committed:
tidy

jdillon committed:
tidy

jdillon committed:
Add configuration for sisu-maven-plugin

jdillon committed:
Move restlet DM entries to restlet1x plugin.

jdillon committed:
Remove objenesis DM should be sorted by litmus
